### PR TITLE
Controller tests

### PR DESCRIPTION
**Pull request by Taeir**
# GUI tests

This PR adds tests for the SettingsScreenController and the SinglePlayerController.
In order to test these controllers, a few things had to be changed and added, which is why this pull request is so large.
## Additions
- Added NoAudioFactory for testing in complete solitude and silence.
- Added NoSoundSlimGuiTest for the same reason. Its a SlimGuiTest, without sounds.
- Added revivePlayer in SinglePlayerPlayingField. This moves the responsibility of reviving the player from the controller to the playingfield (and also makes it easier to test).
- Added tests for SinglePlayerController.
- Added tests for SettingsScreenController
- Added initFXMLForTest() method to the aforementioned controllers, so that the fields that normally get loaded from the FXML file, are created so that the tests can work.

_Note that the real reason we need a NoAudioFactory is that travis does not support our sounds, and the GUI tests for the controller might just fire some sounds._
## Removals
- Removed ActionEvent parameters in some button handle methods. They are not used, so we might as well remove them.
- Removed unused methods
## Changes
- Made GuiTest and SlimGuiTest abstract, so that PMD does not complain about them not containing test methods.
- Refactored the SettingsScreen quite a bit. The code is now (hopefully) easier to read, and makes more use of bindings.
- Renamed some methods, so that their names are easier to understand
- Updated some documentation.
## Reviewing

I advise the following review order:
1. Take a look at the NoAudioFactory, NoSoundSlimGuiTest, SlimGuiTest and GuiTest changes. They are quite easy to understand.
2. Look at the changes to the SinglePlayerController and SinglePlayerPlayingField. It might look like a lot of changes, but thats mostly due to method renaming.
3. Look at the SinglePlayerPlayingFieldTest, I added a test for the new revivePlayer.
4. Look at the SinglePlayerControllerTest
5. Look at the changes to the SettingsScreenController. This one is a bit more complex.
6. Look at the SettingsScreenControllerTest

<feature, testing, layout, naming, documentation

### CODE REVIEW

File: src/main/java/com/github/fishio/control/SinglePlayerController.java [hunk 1]
```diff
@@ -190,25 +181,22 @@ public void onPause(ActionEvent event) {
 			try {
 				playingField.stopGameAndWait();
 			} catch (InterruptedException ex) { }
```
Comment (Diocruel) [inline src/main/java/com/github/fishio/control/SinglePlayerController.java]: _Auto-generated comment_

```
Must have at least one statement.
```

<layout

**Comment by Diocruel** [comment]
Well I think this pr improves the code structure a lot and the extra coverage is welcome! Maybe fix that one checkstyle warning just to reduce the number of warnings throughout our project.

Good to merge after.

<layout, testing, organisation

File: src/main/java/com/github/fishio/control/SettingsScreenController.java [hunk 2]
```diff
@@ -20,34 +21,31 @@
 import javafx.scene.layout.HBox;
 
 /**
- * Help screen controller.
+ * Controller for the settings screen.
```
Comment (ChielBruin) [inline src/main/java/com/github/fishio/control/SettingsScreenController.java:24]: :-) oops my fault!

<naming

File: src/main/java/com/github/fishio/control/SettingsScreenController.java [hunk 3]
```diff
@@ -126,60 +153,90 @@ private int showKeySettings(int row) {
...
-				}
+				//Play the button sound
+				AudioEngine.getInstance().playButtonSound();
+				
+				//The new value is the reverse of the old value
+				boolean nValue = !settings.getBoolean(key);
+				settings.setBoolean(key, nValue);
+				
+				//Update the button text
+				button.setText(toButtonText(nValue));
```
Comment (ChielBruin) [inline src/main/java/com/github/fishio/control/SettingsScreenController.java:184]: Nice Addition. This makes it way more clean!
<logic

**Comment by LMiljak** [comment]
I think this is good to merge 
:+1: 

but pleeeeeaaasssee split up these pull requests a little bit more.

<feature, testing, layout, naming, documentation

**Comment by jessearens** [comment]
With that checkstyle error fixed, it's good to merge.

<layout

**Pull request merged**